server/src/test/java/org/opensearch/index/fielddata/AbstractStringFieldDataTestCase.java

```diff
@@ -340,7 +341,12 @@ public void testSortMissing(boolean first, boolean reverse) throws IOException {
             randomBoolean() ? numDocs : randomIntBetween(10, numDocs),
             new Sort(sortField)
         );
-        assertEquals(numDocs, topDocs.totalHits.value);
+        // As of Lucene 9.0.0, totalHits may be a lower bound
+        if (topDocs.totalHits.relation == TotalHits.Relation.EQUAL_TO) {
+            assertEquals(numDocs, topDocs.totalHits.value);            
+        } else {
+            assertTrue(numDocs >= topDocs.totalHits.value);                        
```

Comment:
Can't we just have 
```
assertTrue(numDocs >= topDocs.totalHits.value); 
```
rather the whole confition?

Reply:
In the test, our total documents (`numDocs`) is not always over 1000, and so it is more precise to test for equality in the 1/3 of cases where a more accurate result will apply.  
https://github.com/opensearch-project/OpenSearch/blob/05a58192433f6e007ade61b4b4f21d0f068d50e0/server/src/test/java/org/opensearch/index/fielddata/AbstractStringFieldDataTestCase.java#L261

This will protect against a bug where `totalHits` is always some small constant like 1, which would pass the lower bound test.

Reply:
(I might argue we could add another assertion that `totalHits.value >= 1000` in this case, but that might be overkill.)

Reply:
Agree it's overkill...